xendit-android/src/main/java/com/xendit/Xendit.java

```diff
@@ -405,22 +407,43 @@ public void onFailure(NetworkError error) {
         }));
     }
 
+    public void get3DSRecommendation(String tokenId, final TokenCallback callback, final Authentication authentication){
+        if (tokenId != null && !tokenId.isEmpty()) {
+            callback.onError(new XenditError("Invalid Token ID"));
+        }
+
+        _get3DSRecommendation(tokenId, new NetworkHandler<ThreeDSRecommendation>().setResultListener(new ResultListener<ThreeDSRecommendation>(){
+            @Override
+            public void onSuccess (ThreeDSRecommendation rec) {
+                callback.onSuccess(new Token(authentication, rec));
+            }
+
+            @Override
+            public void onFailure (NetworkError error) {
+                mLogger.log(Logger.Level.ERROR,  error.responseCode + " " + error.getMessage());
```

Comment:
let's put "3DS Recommendation Error: " before responseCode so that we can identify this is 3DS recommendation error